# HG changeset patch
# User alta88@fixall.com
# Date 1690898787 -7200
# Parent  f0151fc9721f6457ac458762a84f1bd54b930604
Bug 1827882 - Fix too long attachment names in the bar (not list).

Authored by Alta88 and contributed to the Betterbird Project.
See https://github.com/Betterbird/thunderbird-patches/blob/main/LICENSE for license details.
All uses require attribution to the Author.

diff --git a/mail/base/content/msgAttachmentView.inc.xhtml b/mail/base/content/msgAttachmentView.inc.xhtml
--- a/mail/base/content/msgAttachmentView.inc.xhtml
+++ b/mail/base/content/msgAttachmentView.inc.xhtml
@@ -1,38 +1,35 @@
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
                           <!-- the message pane consists of 4 'boxes'. Box #4 is the attachment
                                box which can be toggled into a slim or an expanded view -->
-                          <hbox align="center" id="attachmentBar"
+                          <hbox id="attachmentBar"
                                 context="attachment-toolbar-context-menu"
                                 onclick="if (event.button == 0) { toggleAttachmentList(undefined, true); }">
                             <button type="checkbox" id="attachmentToggle"
                                     onmousedown="event.preventDefault();"
                                     onclick="event.stopPropagation();"
                                     oncommand="toggleAttachmentList(this.checked, true);"/>
-                            <hbox align="center" id="attachmentInfo" flex="1">
+                            <hbox id="attachmentInfo">
                               <html:img id="attachmentIcon"
                                         src="chrome://messenger/skin/icons/attach.svg"
                                         alt="" />
                               <label id="attachmentCount"/>
-                              <label id="attachmentName" crop="center" flex="1"
+                              <label id="attachmentName" crop="center"
                                      role="button"
                                      tooltiptext="&openAttachment.tooltip;"
                                      tooltiptextopen="&openAttachment.tooltip;"
                                      onclick="OpenAttachmentFromBar(event);"
                                      ondragstart="attachmentNameDNDObserver.onDragStart(event);"/>
                               <label id="attachmentSize"/>
+                              <spacer/>
                             </hbox>
-                            <!-- Use a very large flex value here so that attachmentCount doesn't take
-                                 up more space than necessary, but still crops itself if there's not
-                                 enough space. -->
-                            <spacer style="flex: 9999 9999;"/>
 
                             <vbox id="attachment-view-toolbox" class="inline-toolbox">
                               <hbox id="attachment-view-toolbar"
                                     class="toolbar themeable-brighttext"
                                     context="attachment-toolbar-context-menu">
                                 <toolbaritem id="attachmentSaveAll"
                                              title="&saveAllAttachmentsButton1.label;">
                                   <toolbarbutton is="toolbarbutton-menu-button" id="attachmentSaveAllSingle"
diff --git a/mail/themes/shared/mail/messageHeader.css b/mail/themes/shared/mail/messageHeader.css
--- a/mail/themes/shared/mail/messageHeader.css
+++ b/mail/themes/shared/mail/messageHeader.css
@@ -267,16 +267,19 @@ mail-tagfield[collapsed="true"] {
   -webkit-box-orient: vertical;
   -webkit-line-clamp: 3;
   overflow-y: auto;
 }
 
 /* ::::: attachment view ::::: */
 
 #attachmentBar {
+  display: grid;
+  grid-template-columns: max-content auto max-content;
+  align-items: center;
   color: var(--layout-color-1);
   background-color: var(--layout-background-1);
   padding: 3px 0;
 }
 
 #attachmentToggle {
   /* Override button appearance */
   appearance: none;
@@ -312,16 +315,22 @@ mail-tagfield[collapsed="true"] {
   display: none;
 }
 
 #attachmentToggle:focus-visible {
   outline: var(--focus-outline);
   outline-offset: -2px;
 }
 
+#attachmentInfo {
+  display: grid;
+  grid-template-columns: max-content max-content auto max-content 1fr;
+  align-items: center;
+}
+
 #attachmentName:hover,
 #attachmentName[selected="true"] {
   cursor: pointer;
   color: var(--selected-item-text-color);
   background-color: var(--selected-item-color);
 }
 
 #attachmentSize {
@@ -340,17 +349,16 @@ mail-tagfield[collapsed="true"] {
   margin: 0;
   padding: 2px 0;
   margin-inline: 2px 1px;
 }
 
 #attachmentName {
   -moz-user-focus: normal;
   margin: 0;
-  margin-inline-end: -3px;
   padding: 2px 3px;
   border-radius: 2px;
 }
 
 #attachmentName:focus-visible {
   outline: var(--focus-outline);
   outline-offset: -1px;
 }
